cpp/src/Glacier2Lib/NullPermissionsVerifier.cpp

```diff
@@ -26,114 +26,88 @@ class NullSSLPermissionsVerifier : public Glacier2::SSLPermissionsVerifier
-    Identity _nullSSLPVId;
-};
+namespace Glacier2Internal
+{
 
-Init::Init(const CommunicatorPtr& communicator, const string& category, const vector<string>& props) :
-    _communicator(communicator)
+void
+setupNullPermissionsVerifier(
+    const CommunicatorPtr& communicator,
+    const string& category,
+    const vector<string>& permissionsVerifierPropertyNames)
 {
-    _nullPVId.name = "NullPermissionsVerifier";
-    _nullPVId.category = category;
+    const Ice::Identity nullPermissionsVerifierId { "NullPermissionsVerifier", category };
+    const Ice::Identity nullSSLPermissionsVerifierId { "NullSSLPermissionsVerifier",  category };
 
-    _nullSSLPVId.name = "NullSSLPermissionsVerifier";
-    _nullSSLPVId.category = category;
+    Ice::PropertiesPtr properties = communicator->getProperties();
+    ObjectAdapterPtr adapter;
 
-    Ice::PropertiesPtr properties = _communicator->getProperties();
-    for(vector<string>::const_iterator p = props.begin(); p != props.end(); ++p)
+    shared_ptr<Glacier2::PermissionsVerifier> nullPermissionsVerifier;
+    shared_ptr<Glacier2::SSLPermissionsVerifier> nullSSLPermissionsVerifier;
+
+    for (const auto& propertyName : permissionsVerifierPropertyNames)
     {
-        string val = properties->getProperty(*p);
-        if(!val.empty())
+        string val = properties->getProperty(propertyName);
+        if (!val.empty())
         {
-            //
-            // Check permission verifier proxy. It returns a non-empty
-            // value with the new stringified proxy if the property
-            // needs to be rewritten.
-            //
-            val = checkPermissionVerifier(val);
-            if(!val.empty())
+            try
             {
-                properties->setProperty(*p, val);
+                ObjectPrx prx(communicator, val);
+                if (prx->ice_getIdentity() == nullPermissionsVerifierId && nullPermissionsVerifier == nullptr)
+                {
+                    setupObjectAdapter(communicator, adapter);
+                    nullPermissionsVerifier = make_shared<NullPermissionsVerifier>();
+                    adapter->add(nullPermissionsVerifier, nullPermissionsVerifierId);
+                }
+                else if (prx->ice_getIdentity() == nullSSLPermissionsVerifierId && nullSSLPermissionsVerifier == nullptr)
+                {
+                    setupObjectAdapter(communicator, adapter);
+                    nullSSLPermissionsVerifier = make_shared<NullSSLPermissionsVerifier>();
+                    adapter->add(nullSSLPermissionsVerifier, nullSSLPermissionsVerifierId);
+                }
+            }
+            catch(const ProxyParseException&)
+            {
+                // check if it's actually a stringified identity
+                // (with typically missing " " because the category contains a space)
+
+                if(val == communicator->identityToString(nullPermissionsVerifierId) && nullPermissionsVerifier == nullptr)
```

Comment:
I find this code nonsensical. 
If you configure a stringified identity, it should be correct. We should not catch "common mistakes" and auto-fix them.

Reply:
I was unsure, as to why we did this. It would be simpler to just report the ProxyParseException and not try to autofix the properties.